### PR TITLE
Cleanup cache file on Close()

### PR DESCRIPTION
**Pull request by junichif**

### CODE REVIEW

File: gcsstore.go [hunk 1]
```diff
@@ -420,6 +420,7 @@ func (o *gcsFSObject) Close() error {
 	if !o.opened {
 		return nil
 	}
+	defer os.Remove(o.cachepath)
```
Comment (epsniff) [inline gcsstore.go]: I think you need:
```
 func (o *gcsFSObject) Close() (err error) {
  	if !o.opened {		  	if !o.opened {
  		return nil		  		return nil
  	}		  	}
 	defer func(){
             if err != nil {
                  return err
             }
             err = os.Remove(o.cachepath)
       } 


```

**Comment by epsniff** [comment]
Lets plan a future PR to switch all calls in our code to use Close() and remove Release().  I wouldn't mind just cleaning up the API.

**Comment by epsniff** [comment]
:+1: minus the comment above.

**Comment by epsniff** [comment]
Its confusing but `Release()` and `Close()` are different.    Close() tries to Sync() local changes up to GCS, Release() doesn't.

File: gcsstore.go (hunk 1 above)
Comment (junichif) [inline gcsstore.go]: If o.cachedcopy.Sync() or o.cached.Close() fail, don't we still want to clean up the cachepath file, and actually cleanup everything else?

Or do you think, in those cases we'd want to keep the cached file around longer?  

Seems like if Close() is closed we should cleanup everything regardless of where the error happened, so maybe:
```go
func (o *gcsFSObject) Close() error {
	if !o.opened {
		return nil
	}
	defer func() {
		os.Remove(o.cachepath)
		o.cachedcopy = nil
		o.opened = false
	}()

	serr := o.cachedcopy.Sync()
	cerr := o.cachedcopy.Close()
	if serr != nil || cerr != nil {
		return fmt.Errorf("error on sync and closing localfile. %s sync=%v, err=%v", o.cachepath, serr, cerr)
	}

	if o.opened && !o.readonly {
		err := o.Sync()
		if err != nil {
			return err
		}
	}

	return nil
}
```

**Comment by junichif** [comment]
Yeah, I do have a few use cases for Release(); one is when I Open() a file and then there's an error somewhere, then I'll just Release() the file instead of Close()ing it.

**Comment by araddon** [comment]
:+1:

File: gcsstore.go (hunk 1 above)
Comment (epsniff) [inline gcsstore.go]: that seems fine, lets go with this for now.

**Pull request merged**